src/app/oauth/oauth.service.js

```diff
@@ -0,0 +1,137 @@
+//////// Module Definition ////////////
+define([
+	'oauth/oauth.module',
+	'util/util.service'
+], function() {
+///////////////////////////////////////
+angular.module('oauth').factory('oauth', ['util', '$http', '$location', '$route', function(util, $http, $location, $route) {
+	var promise;
+	var dummyLogin = false;
+
+	var setDummyLogin = function(){
+		dummyLogin = true;
+	}
+
+	var unsetDummyLogin = function(){
+		dummyLogin = false;
+	}
+
+
+	var getUserInfo = function(){
+		if (dummyLogin){
+			return Promise.resolve({"userinfo": {"name": "Dummy"}});
+		}
+		if ($location.host() == "localhost"){
+			return Promise.resolve(null);
+		}
+		promise = $http.get($location.protocol() + '://tools.wmflabs.org/widar/?action=get_rights&botmode=1').then(function(response){
```

Comment:
If the parameter botmode=1 is set, Widar returns an json response. Otherwise the response include the html for the complete widar website (where will be some information about the result of the edit as well).